usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateNoOverlappingStopTimeInTripBlock.java

```diff
@@ -0,0 +1,264 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.apache.logging.log4j.Logger;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.Calendar;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.calendardates.CalendarDate;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.calendardates.ExceptionType;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.stoptimes.StopTime;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.trips.Trip;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.error.BlockTripsWithOverlappingStopTimesNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsDataRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+import org.mobilitydata.gtfsvalidator.usecase.utils.TimeUtils;
+
+import java.util.*;
+
+/**
+ * Use case to validate that for a given `trip_id` there is no overlapping {@code StopTime}.
+ */
+public class ValidateNoOverlappingStopTimeInTripBlock {
+    private final GtfsDataRepository dataRepo;
+    private final ValidationResultRepository resultRepo;
+    private final Logger logger;
+    private final TimeUtils timeUtils;
+
+    public ValidateNoOverlappingStopTimeInTripBlock(final GtfsDataRepository dataRepo,
+                                                    final ValidationResultRepository resultRepo,
+                                                    final Logger logger,
+                                                    final TimeUtils timeUtils) {
+        this.dataRepo = dataRepo;
+        this.resultRepo = resultRepo;
+        this.logger = logger;
+        this.timeUtils = timeUtils;
+    }
+
+    /**
+     * Use case execution method: checks for each trip of the same block that no stop times overlap:
+     * - trips from same block with same service id that overlap should generate a notice
+     * - trips from same block with same service_id that do not overlap should not generate notice
+     * - trips from same block with with different service_id that overlap should generate notice whether calendar or
+     * calendar_Dates is provided
+     * - trips from same block with with different service_id that do not overlap should not generate notice whether
+     * calendar or calendar_dates is provided
+     */
+    public void execute() {
+        logger.info("validating rule 'E054 - Trips from same block overlap'");
+
+        final Map<String, List<Trip>> tripPerBlockId = dataRepo.getAllTripByBlockId();
+        final boolean areCalendarProvided = dataRepo.getCalendarAll().size() != 0;
+
+        tripPerBlockId.forEach((blockId, tripCollectionFromSameBlockId) -> {
+            final List<String> visitedTripIdCollection = new ArrayList<>();
+            tripCollectionFromSameBlockId.forEach(currentTrip -> {
```

Comment:
I would try to shorten some of these variable names, as it becomes difficult to read. Maybe just `trips` instead of `tripCollectionFromSameBlockId`?